container/container.go

```diff
@@ -200,7 +203,12 @@ func (container *Container) toDisk() (*Container, error) {
 
 // CheckpointTo makes the Container's current state visible to queries, and persists state.
 // Callers must hold a Container lock.
-func (container *Container) CheckpointTo(store *ViewDB) error {
+func (container *Container) CheckpointTo(ctx context.Context, store *ViewDB) error {
+	_, span := otel.Tracer("").Start(ctx, "daemon.CheckpointTo", trace.WithAttributes(
```

Comment:
```suggestion
	ctx, span := otel.Tracer("").Start(ctx, "daemon.CheckpointTo", trace.WithAttributes(
```
You should always shadow the `ctx` argument. Otherwise it would be too easy to overlook this line when modifying the function body to actually use the context. And it would be too easy for such a mistake to be missed in code review, hidden under a collapsed code fold. If the wrong context gets passed into a call, traces won't match up with the actual call stacks.

[It's not a compile error](https://go.dev/play/p/LjgwPVFIk_E) to reassign an argument variable in a `:=` multi-assignment. Did you write it this way because of some linter? We need to fix the linter in this case, because it is encouraging a footgun.